### DISCUSSION TITLE
Is there an embed functionality for directories?

### DISCUSSION THREAD

**Discussion by serge-hulne** [vlang/v#17177] [Questions and Answers]
Is there an embed functionality for directories?

Could be very useful for compiling a web app as a single executable desktop app (as is done with Go/Wails).

**Comment by larpon** [comment]
Nope, there's only `$embed_file` because V is normally very explicit and embedding whole directories can have some nasty security issues and other implications (like if you accidentially embed your `movies` folder) so if you *really* need to embed complete dirs you could build a V script that generate the `$embed_file` calls of files you want to embed.

**Reply by tobiasjc** [reply]
Hi, I have the same problem as when trying to do a fairly customizable by non-technical people piece of software (configuration files), that is meant to have a single executable deployment (embedded data, known at compile time). Would you please point me to some resources where I can learn how to properly use VScript to embed multiple files of a directory into my application? Thanks in advance!

**Reply by larpon** [reply]
This is a bare bones self contained example [embed_directory_example.zip](https://github.com/vlang/v/files/12857429/embed_directory_example.zip).

**Please note that it is a script I  did in 5 min. so it is not accounting for every edge-case available on the planet use at your own risk**

Instructions:

1. Download and unzip the archive
2. `cd embed_directory_example` change working directory to the unzipped content
3. Observe the contents (`embed_me_1.txt`,`embed_me_2.boo`,`gen_embeds_from_dir.vsh`,`use_embedded_files.v`)
4. Run `v run gen_embeds_from_dir.vsh` to generate the embedding source code
5. Run `v run .` to run the application, using the generated file
6. Observe output and/or *generated* content in `auto_generated_embedded_files.v`

**Reply by larpon** [reply]
I also made a gist for people finding that more useful:
https://gist.github.com/Larpon/110dbfb13a6a795de387e429e81fc039
